README.md

```diff
@@ -220,4 +226,38 @@ Not all steps need to be followed on the instructions for the aad-pod-identity p
     secret1
     ```
 
+#### OPTION 3: VMSS User Assigned Managed Identity
+
+This option allows flexvol to use the user assigned managed identity on the k8s cluster VMSS directly.
+
+### Prerequisites:
+
+Make sure you have AKS with [managed identities](https://docs.microsoft.com/en-us/azure/aks/use-managed-identity) 
+
+1. Get the user identity that created automaticaly by AKS
+
+```bash
+az aks show -g <resource-group> -n <aks_cluster_name> -o json --query "identityProfile"
+```
+
+1. Grant Azure Managed Identity KeyVault permissions
+
+   Ensure that your Azure Identity has the role assignments required to see your Key Vault instance and to access its content. Run the following Azure CLI commands to assign these roles if needed:
+
+   ```bash
+   # set policy to access keys in your Key Vault
+   az keyvault set-policy -n $KV_NAME --key-permissions get --spn <YOUR AZURE MANAGED IDENTITY CLIENT ID>
+   # set policy to access secrets in your Key Vault
+   az keyvault set-policy -n $KV_NAME --secret-permissions get --spn <YOUR AZURE MANAGED IDENTITY CLIENT ID>
+   # set policy to access certs in your Key Vault
+   az keyvault set-policy -n $KV_NAME --certificate-permissions get --spn <YOUR AZURE MANAGED IDENTITY CLIENT ID>
+   ```
+
+2. Deploy your application. Specify `usevmmanagedidentity` to `true`.
+
+```yaml
+usevmmanagedidentity: "true"            # [OPTIONAL] if not provided, will default to "false"
+userAssignedIdentityID: "clientid"      # [OPTIONAL] use the client id to specify which user assigned managed identity to use, leave empty to use system assigned managed identity
+```
```

Comment:
can we also add a step to assign the managed identity to VMSS in case of non AKS clusters?